oceannavigator/frontend/src/components/plot-components/AreaWindow.jsx

```diff
@@ -66,8 +66,14 @@ const AreaWindow = (props) => {
     }
   );
 
+  const isFirstRender = useRef(true);
   // Sync scale when dataset_0.variable changes
   useEffect(() => {
+    //guard useEffect on first render. to handle permalink
+    if (isFirstRender.current) {
+    isFirstRender.current = false;
+    return;
+  }
```

Comment:
Calling `useEffect` with an empty dependency array like
```
useEffect(() => {
 // set state from props.init here
}, [ ] )
```

will only run on the first render should help you avoid that first render ref switch. This would probably be a clean approach to take for all of the plot windows.